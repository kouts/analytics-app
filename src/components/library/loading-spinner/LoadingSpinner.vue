<template>
  <div v-if="show" class="loader" :style="styles"></div>
</template>

<script>
export default {
  props: {
    show: {
      type: Boolean,
      default: false
    },
    size: {
      type: [Number, String],
      default: 32
    },
    width: {
      type: [Number, String],
      default: 4
    }
  },
  computed: {
    styles() {
      return {
        borderWidth: `${Number(this.width)}px`,
        width: `${Number(this.size)}px`,
        height: `${Number(this.size)}px`
      }
    }
  }
}
</script>

<style lang="scss" scoped>
.loader {
  border-radius: 50%;
  width: 32px;
  height: 32px;
  position: relative;
  text-indent: -9999em;
  border-top: 4px solid rgba($gray-400, 0.2);
  border-right: 4px solid rgba($gray-400, 0.2);
  border-bottom: 4px solid rgba($gray-400, 0.2);
  border-left: 4px solid $primary;
  transform: translateZ(0);
  animation: load 0.9s infinite linear;
}
@keyframes load {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
</style>
