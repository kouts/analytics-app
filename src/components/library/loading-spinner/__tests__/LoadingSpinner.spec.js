import LoadingSpinner from '../LoadingSpinner.vue'
import { mount } from '@vue/test-utils'

describe('Loading Spinner', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(LoadingSpinner)
  })

  it('should be defined', () => {
    expect(LoadingSpinner).toBeDefined()
  })

  it('should show depending on the "show" prop', async () => {
    await wrapper.setProps({
      show: true
    })
    expect(wrapper.find('div.loader').exists()).toBe(true)
  })

  it('should hide depending on the "show" prop', async () => {
    await wrapper.setProps({
      show: false
    })
    expect(wrapper.find('div.loader').exists()).toBe(false)
  })

  it('should render with the right size', async () => {
    await wrapper.setProps({
      show: true,
      width: 4,
      size: 32
    })
    expect(wrapper.find('div.loader').attributes().style).toBe('border-width: 4px; width: 32px; height: 32px;')
  })

  describe('snapshots', () => {
    it('should render a loading spinner', async () => {
      await wrapper.setProps({ show: true })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
