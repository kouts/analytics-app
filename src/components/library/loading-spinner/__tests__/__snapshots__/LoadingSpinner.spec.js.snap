// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Spinner snapshots should render a loading spinner 1`] = `
<div
  class="loader"
  style="border-width: 4px; width: 32px; height: 32px;"
/>
`;
